test/e2e/network/service.go

```diff
@@ -2687,12 +2691,12 @@ var _ = SIGDescribe("Services", func() {
 		})
 
 		serviceAddress := net.JoinHostPort(serviceName, strconv.Itoa(port))
-		framework.Logf("waiting up to %v to connect to %v", e2eservice.KubeProxyEndpointLagTimeout, serviceAddress)
+		framework.Logf("waiting up to %v to connect to %v", e2eservice.KubeProxyLagTimeout, serviceAddress)
```

Comment:
this test wasn´t failing and it takes less than 1 minutes in the 5k cluster job, are you sure you want to change it?
```
	ginkgo.It("should be rejected when no endpoints exist", func() {
```

do you want to modify it?

Reply:
It failed here: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-scale-correctness/1351137755316359168 actually.

Reply:
oh, right, but it seems it was something dns related
> + /agnhost connect '--timeout=3s' no-pods:80
> DNS: lookup no-pods on 10.0.0.10:53: no such host
> command terminated with exit code 1

in this case the timeout to get the endpoints should not be important, because kube-proxy has to insert a REJECT rule until it has endpoints for that service ... 
can we do this as a separate commit/PR instead? 
the downside is that kube-proxy alternatives that doesn´t implement this behaviour will take 5 mins to fail, and I know at least 2 CNIs that fail this test ;)

Reply:
I dont think it's obvious it's unrelated. Note that, depending on the call different stderr is reported, e.g.:
```
Jan 18 12:30:38.650: INFO: error didn't contain 'REFUSED', keep trying: error running /workspace/kubernetes/platforms/linux/amd64/kubectl --server=https://34.75.131.177 --kubeconfig=/workspace/.kube/config --namespace=services-1955 exec execpod-noendpointsh8vxd -- /bin/sh -x -c /agnhost connect --timeout=3s no-pods:80:
Command stdout:

stderr:
+ /agnhost connect '--timeout=3s' no-pods:80
OTHER: dial tcp: lookup no-pods on 10.0.0.10:53: no such host
command terminated with exit code 1

error:
exit status 1
```
...
```
Jan 18 12:31:01.217: INFO: error didn't contain 'REFUSED', keep trying: error running /workspace/kubernetes/platforms/linux/amd64/kubectl --server=https://34.75.131.177 --kubeconfig=/workspace/.kube/config --namespace=services-1955 exec execpod-noendpointsh8vxd -- /bin/sh -x -c /agnhost connect --timeout=3s no-pods:80:
Command stdout:

stderr:
+ /agnhost connect '--timeout=3s' no-pods:80
OTHER: dial tcp: lookup no-pods on 10.0.0.10:53: no such host
command terminated with exit code 1

error:
exit status 1
Jan 18 12:31:02.651: INFO: Running '/workspace/kubernetes/platforms/linux/amd64/kubectl --server=https://34.75.131.177 --kubeconfig=/workspace/.kube/config --namespace=services-1955 exec execpod-noendpointsh8vxd -- /bin/sh -x -c /agnhost connect --timeout=3s no-pods:80'
Jan 18 12:31:06.156: INFO: rc: 1
Jan 18 12:31:06.157: INFO: error didn't contain 'REFUSED', keep trying: error running /workspace/kubernetes/platforms/linux/amd64/kubectl --server=https://34.75.131.177 --kubeconfig=/workspace/.kube/config --namespace=services-1955 exec execpod-noendpointsh8vxd -- /bin/sh -x -c /agnhost connect --timeout=3s no-pods:80:
Command stdout:

stderr:
+ /agnhost connect '--timeout=3s' no-pods:80
TIMEOUT
command terminated with exit code 1

error:
exit status 1
```

So clearly something changed 28s in the test - having 30s timeout seems visibly to small in that case.

Reply:
>  that doesn´t implement this behaviour will take 5 mins to fail, and I know at least 2 CNIs that fail this test ;)

Can we detect what CNI we use at the beginning of the test and disable this test for those cases?

Reply:
> > that doesn´t implement this behaviour will take 5 mins to fail, and I know at least 2 CNIs that fail this test ;)
> 
> Can we detect what CNI we use at the beginning of the test and disable this test for those cases?

don´t worry about that, those are downstream CIs, if you add as an independent commit it is cleaner to add a revert downstream with only this test.

Coming back to the test, the intent is that if there are no endpoints you receive a reject so the connection to the service IP doesn´t hang, hijacking the example from https://blog.cloudflare.com/when-tcp-sockets-refuse-to-die/

```
State    Recv-Q Send-Q Local:Port Peer:Port
SYN-RECV 0      0      host:1     host:2    timer:(on,996ms,0)

00:01.033 IP host.1 > host.2: Flags [S.] # first retry
00:03.045 IP host.1 > host.2: Flags [S.] # second retry
00:07.301 IP host.1 > host.2: Flags [S.] # third retry
00:15.493 IP host.1 > host.2: Flags [S.] # fourth retry
00:31.621 IP host.1 > host.2: Flags [S.] # fifth retry
01:04:610 SYN-RECV disappears
```

I think that most distros wait ~1 min, 5 retries by default, if we test that you can add a reject later than 1 minute the feature seems useless ... 
however, what I think we shouldn´t do is to test DNS here, and we should probe the ClusterIP directly, I closed a PR to do that https://github.com/kubernetes/kubernetes/pull/96921 but seems a better fix to add here instead of increasing the timeout.

Reply:
OK - let's start with fixing that other test. And we can come back to this one later.

Regarding using DNS - I'm not convinced. DNS should be working too so if that doesn't work it's also a bad signal.